app/services/process_feed_service.rb

```diff
@@ -22,268 +22,285 @@ def process_entries(xml, account)
   class ProcessEntry
     def call(xml, account)
       @account = account
-      @xml     = xml
-
-      return if skip_unsupported_type?
-
-      case verb
-      when :post, :share
-        return create_status
-      when :delete
-        return delete_status
-      end
+      @fetched = Activity.new(xml)
+
+      return unless [:activity, :note, :comment].include?(@fetched.type)
+
+      klass = case @fetched.verb
```

Comment:
I feel like this could be a private method that is just the switch statement. The indented assignment always looks out of place.